lte/gateway/c/core/oai/include/proto_map.hpp

```diff
@@ -0,0 +1,248 @@
+  }
+  char* get_name() { return name; }
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    get()                                                         **
+  **                                                                        **
+  ** Description: Takes key and valueP as parameters.If the key exists,     **
+  **              corresponding value is returned through the valueP,       **
+  **              else returns error.                                       **
+  **                                                                        **
+  ***************************************************************************/
+
+  proto_map_rc_t get(const keyT key, valueT* valueP) {
+    if (map->empty()) {
+      return PROTO_MAP_EMPTY;
+    }
+    if (valueP == nullptr) {
+      return PROTO_MAP_BAD_PARAMETER_VALUE;
+    }
+    auto search_result = map->find(key);
+    if (search_result != map->end()) {
+      *valueP = search_result->second;
+      return PROTO_MAP_OK;
+    } else {
+      return PROTO_MAP_KEY_NOT_EXISTS;
+    }
+  }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    insert()                                                      **
+  **                                                                        **
+  ** Description: Takes key and value as parameters.Inserts the <key,value> **
+  **              pair into the map. If identical key already exists then   **
+  **              returns error.                                            **
+  **                                                                        **
+  ***************************************************************************/
+  proto_map_rc_t insert(const keyT key, const valueT value) {
+    typedef typename google::protobuf::Map<keyT, valueT>::iterator itr;
+    std::pair<itr, bool> insert_response =
+        map->insert(google::protobuf::MapPair<keyT, valueT>(key, value));
+    if (insert_response.second) {
+      return PROTO_MAP_OK;
+    } else {
+      return PROTO_MAP_KEY_ALREADY_EXISTS;
+    }
+  }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    remove()                                                      **
+  **                                                                        **
+  ** Description: Takes key parameter.Removes the corresponding entry from  **
+  **              the map. If key does not exists returns error             **
+  **                                                                        **
+  ***************************************************************************/
+  proto_map_rc_t remove(const keyT key) {
+    if (map->empty()) {
```

Comment:
Can we clear the map if its empty?

Reply:
In this function, an entry matching to the key needs to be removed. If there no entries in map, then no need to clear the map.